pkg/clusteragent/admission/mutate/autoinstrumentation/auto_instrumentation.go

```diff
@@ -308,15 +307,59 @@ func getPinnedLibraries(registry string) []libInfo {
 	return res
 }
 
-// getLibrariesLanguageDetection runs process language auto-detection and returns languages to inject for APM Instrumentation.
-// The languages information is available in workloadmeta-store and attached on the pod's owner.
-func (w *Webhook) getLibrariesLanguageDetection(pod *corev1.Pod) []libInfo {
-	if config.Datadog().GetBool("admission_controller.auto_instrumentation.inject_auto_detected_libraries") {
-		// Use libraries returned by language detection for APM Instrumentation
-		return w.getAutoDetectedLibraries(pod)
+type libInfoLanguageDetection struct {
+	libs             []libInfo
+	injectionEnabled bool
+}
+
+func (l *libInfoLanguageDetection) containerMutator(v version) containerMutator {
```

Comment:
So if I understand correctly:

- if language detection is not enabled, we inject nothing
- if language detection is enabled we inject what is detected (even if nothing was detecteD) + a flag indicating if injection is enabled or not.

Correct?

Reply:
Yep!